src/Instances/IO/IOMonad.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Marcosh\LamPHPda\Instances\IO;
+
+use Marcosh\LamPHPda\Brand\IOBrand;
+use Marcosh\LamPHPda\HK\HK1;
+use Marcosh\LamPHPda\IO;
+use Marcosh\LamPHPda\Typeclass\Monad;
+
+/**
+ * @implements Monad<IOBrand>
+ *
+ * @psalm-immutable
+ */
+final class IOMonad implements Monad
+{
+    /**
+     * @template A
+     * @template B
+     * @param callable(A): B $f
+     * @param HK1<IOBrand, A> $a
+     * @return IO<B>
+     *
+     * @psalm-pure
+     *
+     * @psalm-suppress LessSpecificImplementedReturnType
+     */
+    public function map(callable $f, HK1 $a): IO
+    {
+        return (new IOFunctor())->map($f, $a);
+    }
+
+    /**
+     * @template A
+     * @template B
+     * @param HK1<IOBrand, callable(A): B> $f
+     * @param HK1<IOBrand, A> $a
+     * @return IO<B>
+     *
+     * @psalm-pure
+     *
+     * @psalm-suppress LessSpecificImplementedReturnType
+     */
+    public function apply(HK1 $f, HK1 $a): IO
+    {
+        return (new IOApply())->apply($f, $a);
+    }
+
+    /**
+     * @template A
+     * @param A $a
+     * @return IO<A>
+     *
+     * @psalm-pure
+     *
+     * @psalm-suppress LessSpecificImplementedReturnType
+     */
+    public function pure($a): IO
+    {
+        return (new IOApplicative())->pure($a);
+    }
+
+    /**
+     * @template A
+     * @template B
+     * @param HK1<IOBrand, A> $a
+     * @param callable(A): HK1<IOBrand, B> $f
+     * @return IO<B>
+     *
+     * @psalm-pure
+     *
+     * @psalm-suppress LessSpecificImplementedReturnType
+     */
+    public function bind(HK1 $a, callable $f): IO
+    {
+        $ioA = IO::fromBrand($a);
+
+        $ioAEval = $ioA->eval();
+
+        /** @var IO<B> $ioAEvalWithF */
+        $ioAEvalWithF = $f($ioAEval);
+
+        $ioAEvalWithFEval = $ioAEvalWithF->eval();
```

Comment:
doing `->eval()` and then `IO::of` should be a no-op, should be enough doing `return IO::fromBrand($f($ioA->eval()))`

Reply:
Done.